src/hooks/resources/device.ts

```diff
@@ -578,3 +594,85 @@ async function checkSupervisorReleaseUpgrades(
 		}
 	}
 }
+
+async function setSupervisorReleaseResource(
+	api: sbvrUtils.PinejsClient,
+	deviceIds: number[],
+	supervisorVersion: string,
+) {
+	if (deviceIds.length === 0) {
+		return;
+	}
+	const devices = (await api.get({
+		resource: 'device',
+		options: {
+			// if the device already has a supervisor_version, just bail.
+			$filter: {
+				id: { $in: deviceIds },
+				supervisor_version: null,
+			},
+			$select: ['id', 'supervisor_version', 'is_of__device_type'],
+		},
+	})) as AnyObject;
+
+	if (devices.length === 0) {
+		return;
+	}
+
+	const devicesByDeviceType = _.groupBy(devices, (d) => {
+		return d.is_of__device_type.__id;
+	});
+
+	if (Object.keys(devicesByDeviceType).length === 0) {
```

Comment:
If you want you could assign this and reuse it for the map like
```ts
const deviceTypes = Object.keys(devicesByDeviceType)
if (deviceTypes.length === 0) {
...
await Promise.all(
	deviceTypes.map(async (deviceType) => {
		const affectedDeviceIds = devicesByDeviceType[deviceType].map((d) => d.id);
```
it's not necessary though